nservicebus/outbox/index.md

```diff
@@ -93,6 +93,10 @@ More detail on each stage of the process:
 9. Update outbox storage to show that the outgoing messages have been sent.
 10. Acknowledge (ACK) receipt of the incoming message so that it is removed from the queue and will not be delivered again.
 
+The key to understanding the Outbox Design Pattern, is realizing that there are two completely distinct processes occurring.  Meaning all of the steps from 1 through 10 are not participating in the same parent transaction or logical process.  In fact, there are two logical processes occurring.  The first process, represents your message handler logic, with the exception that when using the Outbox feature all outbound messages are also serialized and saved in the Outbox table within the same database as your business data.  Which is to imply that your business logic and the serialization and saving of outbound messages occur within the same transaction.  This ensures that the handler logic and all outbound messages are successfully saved or they are all collectively rolled back.  The exception to this would be actions with in your handler that do not enlist in the transient transaction, such as sending emails.  At this juncture, you may be asking yourself, “How do the outbound messages actually get sent if they are stuck serialized in the Outbox table”.  The answer to that question, requires understanding what the second process is responsible for.  Essentially, the second process monitors the Outbox table and does the actual sending of the messages and then updates the Outbox to indicate that it was sent so that it is not picked up and sent again.  It is through these two processes that we are able to simulate guaranteed messaging without the use of distributed transactions, with a few caveats.  In the case of the second process it is technically possible that the same message can be physically sent multiple times.  For example, after a message is sent, an exception may be thrown while trying to update the Outbox to indicate that it was sent.  In this scenario the processing of the Outbox record would be retried and the message would be sent again.  Fortunately, this anomaly is handled by endpoint message handers, thanks to the Deduplication feature which ensures that messages with the same Message Id will only be processed one time, thus making the handlers idempotent. 
```

Comment:
```suggestion
The key to understanding the Outbox Design Pattern is realizing that there are two completely distinct processes occurring.  Meaning all of the steps from 1 through 10 are not participating in the same parent transaction or logical process.  In fact, there are two logical processes occurring.  The first process, represents the message handler logic, with the exception that when using the Outbox feature all outbound messages are also serialized and saved in the Outbox table within the same database as the business data.  Which is to imply that the business logic and the serialization and saving of outbound messages occur within the same transaction.  This ensures that the handler logic and all outbound messages are successfully saved or they are all collectively rolled back.  The exception to this would be, actions within handlers that do not enlist in the transient transaction, such as sending emails.  At this juncture, you may be asking yourself, “How do the outbound messages actually get sent if they are stuck serialized in the Outbox table”.  The answer to that question requires understanding what the second process is responsible for.  Essentially, the second process monitors the Outbox table and does the actual sending of the messages and then updates the Outbox to indicate that it was sent so that it is not picked up and sent again.  It is through these two processes that we are able to simulate guaranteed messaging without the use of distributed transactions, with a few caveats.  In the case of the second process, it is technically possible that the same message can be physically sent multiple times.  For example, after a message is sent, an exception may be thrown while trying to update the Outbox to indicate that it was sent.  In this scenario, the processing of the Outbox record would be retried and the message would be sent again.  Fortunately, this anomaly is handled by endpoint message handers, thanks to the Deduplication feature which ensures that messages with the same MessageId will only be processed one time, thus making the handlers idempotent. 
```